modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PreferenceProperties.java

```diff
@@ -0,0 +1,240 @@
+import javafx.application.Appearance;
+import javafx.beans.InvalidationListener;
+import javafx.beans.property.Property;
+import javafx.beans.property.ReadOnlyObjectProperty;
+import javafx.beans.property.ReadOnlyObjectPropertyBase;
+import javafx.beans.property.ReadOnlyObjectWrapper;
+import javafx.scene.paint.Color;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Contains {@link Property}-based preference implementations.
+ */
+final class PreferenceProperties {
+
+    private final ColorProperty backgroundColor = new ColorProperty("backgroundColor", Color.WHITE);
+    private final ColorProperty foregroundColor = new ColorProperty("foregroundColor", Color.BLACK);
+    private final ColorProperty accentColor = new ColorProperty("accentColor", Color.rgb(21, 126, 251));
+    private final List<ColorProperty> allColors = List.of(backgroundColor, foregroundColor, accentColor);
+    private final AppearanceProperty appearance = new AppearanceProperty();
+    private final Object bean;
+
+    PreferenceProperties(Object bean) {
+        this.bean = bean;
+    }
+
+    public ReadOnlyObjectProperty<Appearance> appearanceProperty() {
+        return appearance.getReadOnlyProperty();
+    }
+
+    public Appearance getAppearance() {
+        return appearance.get();
+    }
+
+    public void setAppearance(Appearance value) {
+        appearance.setValueOverride(value);
+    }
+
+    public ReadOnlyObjectProperty<Color> backgroundColorProperty() {
+        return backgroundColor;
+    }
+
+    public Color getBackgroundColor() {
+        return backgroundColor.get();
+    }
+
+    public void setBackgroundColor(Color color) {
+        backgroundColor.setValueOverride(color);
+    }
+
+    public ReadOnlyObjectProperty<Color> foregroundColorProperty() {
+        return foregroundColor;
+    }
+
+    public Color getForegroundColor() {
+        return foregroundColor.get();
+    }
+
+    public void setForegroundColor(Color color) {
+        foregroundColor.setValueOverride(color);
+    }
+
+    public ReadOnlyObjectProperty<Color> accentColorProperty() {
+        return accentColor;
+    }
+
+    public Color getAccentColor() {
+        return accentColor.get();
+    }
+
+    public void setAccentColor(Color color) {
+        accentColor.setValueOverride(color);
+    }
+
+    public void update(Map<String, ChangedValue> changedPreferences, Map<String, String> wellKnownKeys) {
+        for (Map.Entry<String, ChangedValue> entry : changedPreferences.entrySet()) {
+            String key = wellKnownKeys.get(entry.getKey());
+            if (key != null) {
+                for (ColorProperty colorProperty : allColors) {
+                    if (colorProperty.getName().equals(key)) {
+                        updateColorProperty(colorProperty, entry.getValue().newValue());
+                        break;
+                    }
+                }
+            }
+        }
+
+        fireValueChangedEvent();
```

Comment:
It looks like this will fire `colorProperty.fireValueChangedEvent();` for all colors in `allColors`, even when none has changed.
edit: I see what you did here.  Contrary to it's name, `fireValueChangeEvent` does not always fire.  Perhaps we could rename the method to fireValueChangeEventIfChanged/Maybe or some such?

Reply:
I've renamed it to `fireValueChangedIfNecessary`.